middlewares/accesslog/logger.go

```
@@ -23,11 +29,34 @@ const (
 // Note: Current implementation collects log data but does not have the facility to
 // write anywhere.
 type LogHandler struct {
+	logger *logrus.Logger
+	file   *os.File
 }
 
 // NewLogHandler creates a new LogHandler
-func NewLogHandler() *LogHandler {
-	return &LogHandler{}
+func NewLogHandler(fileName string) (*LogHandler, error) {
+	if len(fileName) > 0 {
+		dir := filepath.Dir(fileName)
+
+		err := os.MkdirAll(dir, 0755)
+		if err != nil {
+			log.Errorf("Failed to create log path %s: %s", dir, err)
```

Shouldn't we return an error here?